### ISSUE TITLE
Support token claims

### ISSUE THREAD

**Issue by espadrine** [espadrine/email-login#5]
In the current API, the basic proof for linking two devices is by verifying an email address, which works by sending a random token by email.

We can link two devices by verifying the token directly.

1. Receive a request for linking a second device to a first device.
2. Send a token to the second device. For instance, as an Aztec code.
3. The token can be transmitted to the first device through photography.
4. Receive the token from the first device.

This proves that whoever controls the second device also controls the first device.

**Comment by jankeromnes** [comment]
I believe this can already be achieved with the undocumented `let claim = session.addClaim('custom-token', '1234secrettoken')`, and `session.proveClaim(claim)` when you're certain the token is legit.

However, I tried this, and my new claim was not actually persisted into the session, but was lost before the next request could retrieve it: Is there a way to `session.save()`?

**Comment by espadrine** [comment]
Yes, but the point of having a defined API boundary is to not think about saving that.

Related: https://www.irccloud.com/pastebin/UXlUvfxN/email-login-claims.txt.
